Library/Formula/telegram.rb

```diff
@@ -0,0 +1,39 @@
+class Telegram < Formula
+  homepage "https://github.com/vysheng/tg"
+  url "https://github.com/vysheng/tg/archive/1.0.5.1.tar.gz"
+  sha1 "72f6a9f89ee9db90b573bad25e567a2015115848"
+
+  depends_on "openssl"
+  depends_on "libconfig"
+  depends_on "readline"
+  depends_on "lua"
+
+  patch :DATA
+
+  def install
+    ENV.append "CFLAGS", "-I/usr/local/include -I#{Formula["readline"].prefix}/include"
+    ENV.append "LDFLAGS", "-L/usr/local/lib -L#{Formula["readline"].prefix}/lib"
+
+    system "./configure", "--prefix=#{prefix}"
+    system "make"
+
+    bin.install "bin/telegram-cli"
+    etc.install "tg-server.pub"
+  end
+
+  test do
+    system "#{bin}/telegram-cli", "-h"
+  end
+end
+
+__END__
+diff --git a/main.c b/main.c
+index ff57ec1..b1d314d 100644
+--- a/main.c
++++ b/main.c
+@@ -804,6 +804,7 @@ int main (int argc, char **argv) {
+   parse_config ();
+
+   tgl_set_rsa_key ("/etc/" PROG_NAME "/server.pub");
++  tgl_set_rsa_key ("/usr/local/etc/tg-server.pub");
```

Comment:
Can't hardcode `/usr/local` - can you use `HOMEBREW_PREFIX` instead? That will be expanded to the location of the user's Homebrew install.